### PR TITLE
Added tranformation warning indicator column

### PR DESCRIPTION
**Pull request by Kluny**
This PR adds a column to wp_edit pages called FBIA Status. The column shows a green light if the post was transformed into a Facebook Instant article with no problems, a yellow light if there were warnings, and a red light if the post was empty after transformation.

There is a checkbox in the FBIA Options page that defaults to false, so that editors can decide whether to show the column or not. The checkbox is placed alongside the "Publish with warnings" option for good grouping of concerns.

A new stylesheet has been added to contain the styling rules for the warning indicators. Since the indicators are very minimal, just red, green, and yellow lights, an informative title text is available on mouseover to help with accessibility.

Relates to #871

### CODE REVIEW

**Comment by Kluny** [comment]
Here's how the column looks in index view. 

image

Here's the option in FBIA settings. 

image

**Comment by adamsilverstein** [comment]
Nice work @Kluny this seems like a super useful feature!

File: facebook-instant-articles.php [hunk 1]
```diff
@@ -439,4 +448,52 @@ function invalidate_all_posts_transformation_info_cache( $option ) {
...
+	function fbia_indicator_column_heading( $columns ) {
+		$publishing_settings = Instant_Articles_Option_Publishing::get_option_decoded();
+		$display_warning_column = $publishing_settings[ 'display_warning_column' ];
+
+		if( "1" === $display_warning_column ) {
+			$columns['FBIA'] = "<span title='Facebook Instant Article Distribution Status' class='fbia-col-heading'>FBIA Status</span>";
+		}
+		return $columns;
+	}
+	add_filter('manage_posts_columns', 'fbia_indicator_column_heading');
```
Comment (adamsilverstein) [inline facebook-instant-articles.php]: missing some whitespace here: add_filter( 'manage_posts_columns', 'fbia_indicator_column_heading' );

File: facebook-instant-articles.php [hunk 2]
```diff
@@ -439,4 +448,52 @@ function invalidate_all_posts_transformation_info_cache( $option ) {
...
+
+			$yellow_light = "<span title='Instant article has warnings after transformation.' class='instant-articles-col-status warning'></span>";
+
+			$green_light = "<span title='Instant article transformed successfully.' class='instant-articles-col-status ok'></span>";
+
+			if ( $column_name === "FBIA" ) {
+				$post = get_post( $post_ID );
+				$instant_articles_post = new \Instant_Articles_Post( $post );
+
+				$is_empty = esc_html( $instant_articles_post->is_empty_after_transformation() );
```
Comment (adamsilverstein) [inline facebook-instant-articles.php]: I don't think you need esc_html here, it is for output

File: facebook-instant-articles.php [hunk 3]
```diff
@@ -439,4 +448,52 @@ function invalidate_all_posts_transformation_info_cache( $option ) {
...
+				if ( "1" === $is_empty ) {
+					echo $red_light;
```
Comment (adamsilverstein) [inline facebook-instant-articles.php]: add wp_kses_post: echo wp_kses_post( $red_light );

File: facebook-instant-articles.php [hunk 4]
```diff
@@ -439,4 +448,52 @@ function invalidate_all_posts_transformation_info_cache( $option ) {
...
+					return;
+				}
+
+				$has_warnings = esc_html( $instant_articles_post->has_warnings_after_transformation() );
```
Comment (adamsilverstein) [inline facebook-instant-articles.php]: remove esc_html

File: facebook-instant-articles.php [hunk 5]
```diff
@@ -439,4 +448,52 @@ function invalidate_all_posts_transformation_info_cache( $option ) {
...
+				if ( "1" === $has_warnings ) {
+					echo $yellow_light;
```
Comment (adamsilverstein) [inline facebook-instant-articles.php]: add wp_kses_post

File: facebook-instant-articles.php [hunk 6]
```diff
@@ -439,4 +448,52 @@ function invalidate_all_posts_transformation_info_cache( $option ) {
...
+					return;
+				}
+
+				echo $green_light;
```
Comment (adamsilverstein) [inline facebook-instant-articles.php]: add wp_kses_post

File: facebook-instant-articles.php [hunk 7]
```diff
@@ -439,4 +448,52 @@ function invalidate_all_posts_transformation_info_cache( $option ) {
...
+
+				return;
+			}
+		}
+	}
+	add_action('manage_posts_custom_column', 'fbia_indication_column', 10, 2);
```
Comment (adamsilverstein) [inline facebook-instant-articles.php]: add whitespace

File: facebook-instant-articles.php [hunk 8]
```diff
@@ -439,4 +448,52 @@ function invalidate_all_posts_transformation_info_cache( $option ) {
...
+		return $columns;
+	}
+	add_filter('manage_posts_columns', 'fbia_indicator_column_heading');
+
+	function fbia_indication_column( $column_name, $post_ID ) {
+		$publishing_settings = Instant_Articles_Option_Publishing::get_option_decoded();
+		$display_warning_column = $publishing_settings[ 'display_warning_column' ];
+
+		if( "1" === $display_warning_column ) {
+			$red_light = "<span title='Instant article is empty after transformation.' class='instant-articles-col-status error.'></span>";
```
Comment (adamsilverstein) [inline facebook-instant-articles.php]: In these html strings, the double quotes should be part of the html and the single quotes in the code, unless i'm missing something... so for this one:
$red_light = '<span title="Instant article is empty after transformation." class="instant-articles-col-status error."></span>';

**Comment by Kluny** [comment]
Made changes as suggested.

**Comment by Kluny** [comment]
Just working on these fixes now.

> Also double check the if checks to properly display colors. On my env tests it was all painted green. 

Here's some HTML that should cause a warning yellow light - use the classic editor in HTML text mode, enter this string and update the post.

`<div class="media-credit-container alignnone"> Some stuff in here </div>`

**Comment by Kluny** [comment]
@vkama, I think you have the ability to merge branches - can this one be merged?

**Comment by adamsilverstein** [comment]
@everton-rosario has your feedback been addressed here? can this be merged?

**Comment by adamsilverstein** [comment]
@everton-rosario @mburak Any additional changes you would like to see here to get this in? 

Can we get this work merged before it becomes stale?

**Comment by everton-rosario** [comment]
Sorry for the delay @adamsilverstein

**Comment by Kluny** [comment]
Thank you @everton-rosario!

**Comment by adamsilverstein** [comment]
Woo hoo! Thanks @everton-rosario.

**Comment by everton-rosario** [comment]
This will be out on the next release. Till there, you can get it from master ;-)

**Pull request merged**